pkg/authorization/api/validation/validation.go

```diff
@@ -212,7 +213,30 @@ func ValidateRole(role *authorizationapi.Role, isNamespaced bool) field.ErrorLis
 }
 
 func validateRole(role *authorizationapi.Role, isNamespaced bool, fldPath *field.Path) field.ErrorList {
-	return validation.ValidateObjectMeta(&role.ObjectMeta, isNamespaced, oapi.MinimalNameRequirements, fldPath.Child("metadata"))
+	allErrs := validation.ValidateObjectMeta(&role.ObjectMeta, isNamespaced, oapi.MinimalNameRequirements, fldPath.Child("metadata"))
+	for i, rule := range role.Rules {
+		allErrs = append(allErrs, validatePolicyRule(rule, fldPath.Child("rules").Index(i))...)
+	}
+
+	return allErrs
+}
+
+func validatePolicyRule(rule authorizationapi.PolicyRule, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(rule.Selector) != 0 {
```

Comment:
I think you want to compare this against nil instead? Otherwise, an non-nil empty map won't be validated (so you will never hit the second error here).